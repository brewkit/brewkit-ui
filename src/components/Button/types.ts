import * as React from 'react';
import { Theme, ThemeOptions } from '@material-ui/core/styles/createMuiTheme';
import { ButtonProps } from '@material-ui/core/Button';


/**
 * Declare any theme variables we want to make available.
 */
declare module '@material-ui/core/styles/createMuiTheme' {
    interface Theme {
        BkButton: {
            loading: React.CSSProperties,
        },
    }
    // allow configuration using `createMuiTheme`
    interface ThemeOptions {
        BkButton?: {
            loading?: React.CSSProperties,
        },
    }
}


export type Props = ButtonProps & {

    /**
     * If `true`, the button will be disabled and show a spinner.
     * @default false
     */
    loading?: boolean,


    /**
     * @ignore
     */
    classes?: Record<string, any>,

};
