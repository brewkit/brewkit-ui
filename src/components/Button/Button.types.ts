import * as React from 'react';
import { ButtonProps as MUIButtonProps } from '@material-ui/core';


/**
 * Declare any theme variables we want to make available.
 */
declare module '@material-ui/core/styles/createTheme' {
    interface Theme {
        BkButton: {
            loading: React.CSSProperties,
        },
    }
    // allow configuration using `createTheme`
    interface ThemeOptions {
        BkButton?: {
            loading?: React.CSSProperties,
        },
    }
}


export type ButtonProps = MUIButtonProps & {

    /**
     * If `true`, the button will be disabled and show a spinner.
     * @default false
     */
    loading?: boolean,


    /**
     * @ignore
     */
    classes?: Record<string, any>,

};
