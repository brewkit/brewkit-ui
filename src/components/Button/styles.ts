import { Theme } from '@material-ui/core/styles/createMuiTheme';


const styles = (theme: Theme): any => ({

    root: {
        padding: '1em 1.25em',
    },

    content: {
        position: 'relative',
        top: 0,
        display: 'flex',
        alignItems: 'center',
        justifyContent: 'center',
        height: '1em',
        opacity: 1,
        transition: 'all 0.3s',

        '& > *:not(:last-child)': {
            marginRight: '0.5rem',
        },
    },

    loader: {
        pointerEvents: 'none',
        opacity: 0,
        position: 'absolute',
        top: 'calc(50% + 1rem)',
        left: '50%',
        transform: 'translate(-50%, -50%)',
        transition: 'all 0.3s',
    },

    loading: {
        opacity: theme?.BkButton?.loading?.opacity ?? 0.5,

        '& $content': {
            top: '-1rem',
            opacity: 0,
        },

        '& $loader': {
            top: 'calc(50% + 2px)',
            opacity: 1,
        },

    },

});


export default styles;
