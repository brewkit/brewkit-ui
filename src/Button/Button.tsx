import * as React from 'react';
import { merge } from 'lodash';
import clsx from 'clsx';
import MUIButton, {
    ButtonClassKey as MUIButtonClassKey,
    ButtonProps as MUIButtonProps,
} from '@material-ui/core/Button';
import { ClassNameMap } from '@material-ui/core/styles/withStyles';
import CircularProgress, { BkCircularProgressProps } from '../CircularProgress';
import { useStyles } from './Button.styles';
import withoutKeys from '../utils/withoutKeys';

type BkButtonClassKey = 'loading' | 'loader' | 'content';

export type BkButtonProps = MUIButtonProps & {
    /**
     * Custom classes for Button
     */
    classes?: ClassNameMap<MUIButtonClassKey | BkButtonClassKey>;

    /**
     * If `true`, the button will be disabled and show a spinner.
     * @default false
     */
    loading?: boolean;

    /**
     * Props for circular progress if loading is present
     * @default {}
     */
    CircularProgressProps?: BkCircularProgressProps;
};

const Button = React.forwardRef(
    (
        props: BkButtonProps,
        ref: React.Ref<HTMLButtonElement>,
    ): React.ReactElement => {
        const {
            loading = false,
            disabled,
            classes: userClasses = {} as BkButtonProps['classes'],
            CircularProgressProps = {} as BkCircularProgressProps,
            disableElevation = true,
            startIcon,
            endIcon,
            children,
            className,
            variant = 'contained',
            color = 'primary',
            ...otherProps
        } = props;

        const classes = useStyles();
        const mergedClasses = merge(
            withoutKeys(classes, ['loader', 'loading', 'content']),
            userClasses,
        );

        return (
            <MUIButton
                {...otherProps}
                className={clsx(
                    classes.root,
                    loading && classes.loading,
                    className,
                )}
                classes={mergedClasses}
                color={color}
                disableElevation={disableElevation}
                disabled={disabled ?? loading}
                ref={ref}
                variant={variant}>
                <div className={clsx(classes?.content, userClasses?.content)}>
                    {startIcon}
                    {children && <span>{children}</span>}
                    {endIcon}
                </div>

                <div className={clsx(classes?.loader, userClasses?.loader)}>
                    <CircularProgress
                        {...CircularProgressProps}
                        color="primary"
                        size="1rem"
                        thickness={5}
                    />
                </div>
            </MUIButton>
        );
    },
);

Button.displayName = 'BkButton';

export default Button;
