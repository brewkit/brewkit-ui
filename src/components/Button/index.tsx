import * as React from 'react';
import clsx from 'clsx';
import { withStyles } from '@material-ui/core/styles';
import MuiButton from '@material-ui/core/Button';
import CircularProgress from '@components/CircularProgress';
import styles from './styles';
import { Props } from './types';


/**
 * Buttons allow users to take actions and make choices with a single tap.
 *
 * [Material-UI Docs](https://material-ui.com/components/buttons/)
 *
 *
 * ## Differences in Brewkit:
 *
 * - Adds a `loading` prop that automatically disables the button and shows a loading state. Loading class can be
 * styled with `BkButton.loading` within your theme.
 */
export const Button = React.forwardRef(({
    loading,
    disabled,
    startIcon,
    endIcon,
    children,
    className,
    classes,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    /**
     * merge our custom classes? into the ones passed to the component
     */
    const buttonClasses = clsx(
        classes?.root,
        loading && classes?.loading,
        className,
    );


    return (
        <MuiButton
            {...otherProps}
            className={buttonClasses}
            disabled={disabled ?? loading}
            ref={ref}
        >

            <div className={classes?.content}>
                {startIcon}
                {children && <span>{children}</span>}
                {endIcon}
            </div>

            <div className={classes?.loader}>
                <CircularProgress color="primary" size="1rem" thickness={5} />
            </div>

        </MuiButton>
    );


});


Button.displayName = 'Button';


export default withStyles(styles)(Button);
