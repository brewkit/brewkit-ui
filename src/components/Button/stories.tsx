import * as React from 'react';
import { withStyles } from '@material-ui/core/styles';
import { text, select, boolean } from '@storybook/addon-knobs';
import ArrowBack from '@material-ui/icons/ArrowBack';
import AddCircleOutline from '@material-ui/icons/AddCircleOutline';


/**
 * This is an annoying hack to get Storybook to properly parse the documentation when the underlying component
 * is wrapped in a HOC. Expect to do/see this on every component with MUI theming options.
 */
import { Button as Component } from './index';
import styles from './styles';
const Button = withStyles(styles)(Component)


export default {
    component: Component,
    title: 'Inputs/Button',
};


export const Sandbox = (): React.ReactElement => {

    const color = select('color', ['default', 'primary', 'secondary', 'inherit'], 'primary');
    const disabled = boolean('disabled', false);
    const disableElevation = boolean('disableElevation', true);
    const fullWidth = boolean('fullWidth', false);
    const loading = boolean('loading', false);
    const variant = select('variant', ['contained', 'outlined', 'text'], 'contained');
    const size = select('size', ['large', 'medium', 'small'], 'medium');
    const children = text('children', 'default');

    return (
        <Button
            color={color}
            disabled={disabled}
            disableElevation={disableElevation}
            fullWidth={fullWidth}
            loading={loading}
            variant={variant}
            size={size}
            children={children}
        />
    );

};


export const General = (): React.ReactElement => (
    <div style={{ display: 'grid', gridAutoFlow: 'column', gridGap: '1rem' }}>
        <Button>Generic Button</Button>
        <Button variant="text">Generic Button</Button>
    </div>
)


export const WithStartIcon = (): React.ReactElement => (
    <Button startIcon={<ArrowBack />}>Generic Button</Button>
)


export const WithEndIcon = (): React.ReactElement => (
    <Button endIcon={<AddCircleOutline />}>Generic Button</Button>
)


export const OnlyIcon = (): React.ReactElement => (
    <Button>
        <AddCircleOutline />
    </Button>
)


export const Disabled = (): React.ReactElement => (
    <div style={{ display: 'grid', gridAutoFlow: 'column', gridGap: '1rem' }}>
        <Button disabled>Generic Button</Button>
        <Button variant="text" disabled>Generic Button</Button>
    </div>
)


export const Loading = (): React.ReactElement => (
    <div style={{ display: 'grid', gridAutoFlow: 'column', gridGap: '1rem' }}>
        <Button loading>Generic Button</Button>
        <Button variant="text" loading>Generic Button</Button>
    </div>
)
