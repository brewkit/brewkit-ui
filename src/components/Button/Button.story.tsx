import * as React from 'react';
import { Story } from '@storybook/react';
import ArrowBack from '@material-ui/icons/ArrowBack';
import AddCircleOutline from '@material-ui/icons/AddCircleOutline';
import { ButtonProps } from './Button.types';

/**
 * This is an annoying hack to get Storybook to properly parse the documentation when the underlying component
 * is wrapped in a HOC. Expect to do/see this on every component with MUI theming options.
 */
import { Button } from './Button';

export default {
    component: Button,
    title: 'Inputs/Button',
};

const defaultArgs = {
    disabled: false,
    disableElevation: true,
    fullWidth: false,
    children: 'I am button text',
};

export const Sandbox: Story<ButtonProps> = function sandbox(args) {
    return <Button {...args} />;
};

Sandbox.args = {
    ...defaultArgs,
    color: 'primary',
};

export const General: Story<ButtonProps> = function storyGeneral(args) {
    return <Button {...args} />;
};

General.args = {
    ...defaultArgs,
    color: 'primary',
    variant: 'text',
};

export const WithStartIcon: Story<ButtonProps> = function storyWithStartIcon(
    args,
) {
    return <Button {...args} />;
};

WithStartIcon.args = {
    ...defaultArgs,
    color: 'primary',
    startIcon: <ArrowBack />,
};

export const WithEndIcon: Story<ButtonProps> = function storyWithEndIcon(args) {
    return <Button {...args} />;
};

WithEndIcon.args = {
    ...defaultArgs,
    color: 'primary',
    endIcon: <AddCircleOutline />,
};

export const OnlyIcon: Story<ButtonProps> = function storyOnlyIcon(args) {
    return <Button {...args} />;
};

OnlyIcon.args = {
    ...defaultArgs,
    color: 'primary',
    children: <AddCircleOutline />,
};

export const Disabled: Story<ButtonProps> = function storyDisabled(args) {
    return <Button {...args} />;
};

Disabled.args = {
    ...defaultArgs,
    color: 'primary',
    disabled: true,
};

export const Loading: Story<ButtonProps> = function storyLoading(args) {
    return <Button {...args} />;
};

Loading.args = {
    ...defaultArgs,
    color: 'primary',
    disabled: true,
    loading: true,
};
