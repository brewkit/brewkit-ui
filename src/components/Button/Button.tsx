import * as React from 'react';
import clsx from 'clsx';

import MuiButton from '@material-ui/core/Button';

import { CircularProgress } from '../CircularProgress';

import useStyles from './Button.styles';
import { ButtonProps } from './Button.types';

/**
 * Buttons allow users to take actions and make choices with a single tap.
 *
 * [Material-UI Docs](https://material-ui.com/components/buttons/)
 *
 *
 * ## Differences in Brewkit:
 *
 * - Adds a `loading` prop that automatically disables the button and shows a loading state. Loading class can be
 * styled with `BkButton.loading` within your theme.
 */
export const Button = React.forwardRef(
    (
        props: ButtonProps,
        ref: React.Ref<HTMLButtonElement>,
    ): React.ReactElement => {
        const {
            loading,
            disabled,
            startIcon,
            endIcon,
            children,
            className,
            ...otherProps
        } = props;

        const classes = useStyles();

        /**
         * merge our custom classes? into the ones passed to the component
         */
        const buttonClasses = clsx(
            classes?.root,
            loading && classes?.loading,
            className,
        );

        return (
            <MuiButton
                {...otherProps}
                className={buttonClasses}
                disabled={disabled ?? loading}
                ref={ref}>
                <div className={classes?.content}>
                    {startIcon}
                    {children && <span>{children}</span>}
                    {endIcon}
                </div>

                <div className={classes?.loader}>
                    <CircularProgress
                        color="primary"
                        size="1rem"
                        thickness={5}
                    />
                </div>
            </MuiButton>
        );
    },
);

Button.displayName = 'Button';

export default Button;
