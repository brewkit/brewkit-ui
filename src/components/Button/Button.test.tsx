// @ts-nocheck
import * as React from 'react';
import { create } from 'react-test-renderer';
import Button from '.';
import { ArrowBack, ArrowBackIosSharp } from '@material-ui/icons';

it('Renders a plain button', () => {
    const comp = create(
        <Button>
            Button with default props
        </Button>
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Renders a button with startIcon prop', () => {
    const comp = create(
        <Button
            startIcon={<ArrowBack />}
        >
            Go back
        </Button>
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Ensure rerender happens on startIcon prop change', () => {
    const comp = create(
        <Button
            startIcon={<ArrowBack />}
        >
            Go back
        </Button>
    );
    let tree = comp.toJSON();

    expect(tree).toMatchSnapshot();

    comp.update(
        <Button
            startIcon={<ArrowBackIosSharp />}
        >
            Go back (IOS)
        </Button>
    );

    tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Renders a button with endIcon prop', () => {
    const comp = create(
        <Button
            endIcon={<ArrowBack />}
        >
            Go back
        </Button>
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Ensure rerender happens on endIcon prop change', () => {
    const comp = create(
        <Button
            endIcon={<ArrowBack />}
        >
            Go back
        </Button>
    );
    let tree = comp.toJSON();

    expect(tree).toMatchSnapshot();

    comp.update(
        <Button
            endIcon={<ArrowBackIosSharp />}
        >
            Go back (IOS)
        </Button>
    );

    tree = comp.toJSON();
    expect(tree).toMatchSnapshot();
});

it('Renders a button with loading prop', () => {
    const comp = create(
        <Button loading />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});

it('Renders a button with disabled prop', () => {
    const comp = create(
        <Button disabled />
    ).toJSON();

    expect(comp).toMatchSnapshot();
});