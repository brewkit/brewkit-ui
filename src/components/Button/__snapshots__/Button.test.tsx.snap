// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ensure rerender happens on endIcon prop change 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <span>
        Go back
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Ensure rerender happens on endIcon prop change 2`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <span>
        Go back (IOS)
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Ensure rerender happens on startIcon prop change 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span>
        Go back
      </span>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;

exports[`Ensure rerender happens on startIcon prop change 2`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
      <span>
        Go back (IOS)
      </span>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Renders a button with disabled prop 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1 Mui-disabled Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    />
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;

exports[`Renders a button with endIcon prop 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <span>
        Go back
      </span>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;

exports[`Renders a button with loading prop 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1 Button-loading-4 Mui-disabled Mui-disabled"
  disabled={true}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={-1}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    />
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;

exports[`Renders a button with startIcon prop 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 11H7.83l5.59-5.59L12 4l-8 8 8 8 1.41-1.41L7.83 13H20v-2z"
        />
      </svg>
      <span>
        Go back
      </span>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
  <span
    className="MuiTouchRipple-root"
  />
</button>
`;

exports[`Renders a plain button 1`] = `
<button
  className="MuiButtonBase-root MuiButton-root MuiButton-text Button-root-1"
  disabled={false}
  onBlur={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex={0}
  type="button"
>
  <span
    className="MuiButton-label"
  >
    <div
      className="Button-content-2"
    >
      <span>
        Button with default props
      </span>
    </div>
    <div
      className="Button-loader-3"
    >
      <div
        className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
        role="progressbar"
        style={
          Object {
            "height": "1rem",
            "width": "1rem",
          }
        }
      >
        <svg
          className="MuiCircularProgress-svg"
          viewBox="22 22 44 44"
        >
          <circle
            className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
            cx={44}
            cy={44}
            fill="none"
            r={19.5}
            strokeWidth={5}
            style={Object {}}
          />
        </svg>
      </div>
    </div>
  </span>
</button>
`;
